validator/sawtooth_validator/execution/executor.py

```diff
@@ -348,9 +348,6 @@ def _send_and_process_result(self, content, connection_id, signature):
                 {signature: fut}
 
     def remove_broken_connection(self, connection_id):
-        if connection_id not in self._open_futures:
```

Comment:
This guard still seems like it could be necessary.

Reply:
with this check in place the issue is not resolved. i tested again today (on 1.0.4) without this check  issue reported in this bug persists.

Reply:
@mithunshashidhara it's possible that the actual bug then is in the management of that list. I don't think we want to remove a check if we don't understand why it's resolving a symptom.

Reply:
there is no need to send response to TP as it has been stopped.

Reply:
Can you explain how this was tested? Was it tested with both schedulers? I am concerned with the fact that futures are not being resolved and no InternalError is being returned. As such the txn will not be retryed and the serial scheduler will hang waiting on that future. I will try to test this as well but I am curious what your testing strategy was.

Reply:
switch to 1.0.4, run "./bin/build_all -t python" to build docker images. goto directory docker/compose/  "docker-compose -f sawtooth-local.yaml up" . copy my test script to docker/compose/  and run the script https://jira.hyperledger.org/secure/attachment/15726/STL-1115_repro
with my changes you should see 110,120,130,140,150.
without changes you will see inconsistent result. today i tested with both schedulers and things are fine with my changes.